test/index.spec.ts

```diff
@@ -63,19 +63,31 @@ describe('resin-semver', () => {
 		});
 
 		it('should correctly compare ".dev" versions', () => {
-			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(1);
+			expect(semver.compare('2.0.0', '2.0.0+dev')).to.equal(-1);
 			expect(semver.compare('2.0.0', '2.0.0-dev')).to.equal(1);
-			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(1);
-			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(-1);
+			expect(semver.compare('2.0.0', '2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('2.0.0+rev6', '2.0.0+rev6.dev')).to.equal(-1);
+			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6')).to.equal(1);
 			expect(semver.compare('2.0.0+rev6.dev', '2.0.0+rev6.dev')).to.equal(0);
-			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3', 'Resin OS 2.0.0+rev3.dev')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3')).to.equal(1);
 			expect(semver.compare('Resin OS 2.0.0+rev3.dev', 'Resin OS 2.0.0+rev3.dev')).to.equal(0);
-			expect(semver.compare('2.0.0', 'Resin OS 2.0.0.dev')).to.equal(1);
-			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0')).to.equal(-1);
+			expect(semver.compare('2.0.0', 'Resin OS 2.0.0.dev')).to.equal(1);// B is invalid
+			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0')).to.equal(-1);// A is invalid
 			expect(semver.compare('Resin OS 2.0.0.dev', '2.0.0.dev')).to.equal(0);
 		});
+
+		it('should correctly compare "(dev)" and "(prod)" versions without revisions', () => {
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0 (dev)')).to.equal(-1);
+			expect(semver.compare('Resin OS 2.0.0', 'Resin OS 2.0.0 (prod)')).to.equal(-1);
```

Comment:
Ok, interesting. We should probably bring @imrehg in here, in case there are any specific reasons for this? My understanding is that a version without a variant is essentially a prod version, so should be higher ranked. Is that not true? I'm not sure it'll even come up, but it'd be nice to have a clear rule anyway.

The only case where it could actually matter is in version ranking, when downloading from the dashboard: the highest ranked version is what we'll be recommending to users, and I think `2.0.0`, like `2.0.0 (prod)`, should be more recommended that `2.0.0 (dev)`. We've got a strong preference for avoiding users accidentally using dev versions in production.